src/datadog/tracer_config.cpp

```diff
@@ -119,6 +122,134 @@ Expected<std::unordered_map<std::string, std::string>> parse_tags(
+    throw error->with_prefix(prefix);
+  }
+  return *styles;
+}
+
+std::string json_quoted(StringView text) {
+  std::string unquoted;
+  assign(unquoted, text);
+  return nlohmann::json(unquoted).dump();
+}
+
+Expected<void> finalize_propagation_styles(FinalizedTracerConfig &result,
+                                           const TracerConfig &config,
+                                           Logger &logger) {
+  namespace env = environment;
+  // Print a warning if a questionable combination of environment variables is
+  // defined.
+  const auto ts = env::DD_TRACE_PROPAGATION_STYLE;
+  const auto tse = env::DD_TRACE_PROPAGATION_STYLE_EXTRACT;
+  const auto se = env::DD_PROPAGATION_STYLE_EXTRACT;
+  const auto tsi = env::DD_TRACE_PROPAGATION_STYLE_INJECT;
+  const auto si = env::DD_PROPAGATION_STYLE_INJECT;
+  // clang-format off
+  /*
+           ts    tse   se    tsi   si
+           ---   ---   ---   ---   ---
+    ts  |  x     warn  warn  warn  warn
+        |
+    tse |  x     x     warn  ok    ok
+        |
+    se  |  x     x     x     ok    ok
+        |
+    tsi |  x     x     x     x     warn
+        |
+    si  |  x     x     x     x     x
+  */
+  // In each pair, the first would be overridden by the second.
+  const std::pair<env::Variable, env::Variable> questionable_combinations[] = {
+           {ts, tse}, {ts, se},  {ts, tsi}, {ts, si},
+
+                      {se, tse}, /* ok */   /* ok */
+
+                                 /* ok */   /* ok */
+
+                                            {si, tsi},
+  };
+  // clang-format on
+
+  const auto warn_message = [](StringView name, StringView value,
+                               StringView name_override,
+                               StringView value_override) {
+    std::string message;
+    message += "Both the environment variables ";
+    append(message, name);
+    message += "=";
+    message += json_quoted(value);
+    message += " and ";
+    append(message, name_override);
+    message += "=";
+    message += json_quoted(value_override);
+    message += " are defined. ";
+    append(message, name_override);
+    message += " will take precedence.";
+    return message;
+  };
+
+  for (const auto &[var, var_override] : questionable_combinations) {
+    const auto value = lookup(var);
+    if (!value) {
+      continue;
+    }
+    const auto value_override = lookup(var_override);
+    if (!value_override) {
+      continue;
+    }
+    const auto var_name = name(var);
+    const auto var_name_override = name(var_override);
+    logger.log_error(Error{
+        Error::MULTIPLE_PROPAGATION_STYLE_ENVIRONMENT_VARIABLES,
+        warn_message(var_name, *value, var_name_override, *value_override)});
+  }
+
+  // Parse the propagation styles from the configuration and/or from the
+  // environment.
+  // Exceptions make this section simpler.
+  try {
+    const auto global_styles = styles_from_env(env::DD_TRACE_PROPAGATION_STYLE);
```

Comment:
By this point, I'm wondering the value here of the macro aligning variable names with environment variable names.

Reply:
The benefit is that there's one way to add an environment variable: by adding an enum value.  The list of environment variables consumed by nginx is then automatically up-to-date.